### ISSUE TITLE
Add onError handler for posthog init

### ISSUE THREAD

**Issue by MarkLyck** [PostHog/posthog-js#482]
One major problem with Posthog is that ad blockers are blocking it. Which I'm fine with for analytics. But because I also use posthog for feature flags that is core functionality, that a user would likely want to work.

However right now, if you have an adblocker like the default one in Brave Browser, posthog init will just fail and keep retrying. But there's no way to catch the error and prompt the user that they might want to disable their adblocker.

Please add an onError handler to catch the error when it fails to init for whatever reason.

**Comment by EugeneBalabai** [comment]
Any news? I want to set a default value for my feature flags if an error occurs but I didn't find any solution

**Comment by mtunyk** [comment]
Sending events to PostHog [from your own domain](https://posthog.com/docs/advanced/proxy) should be a workaround of the issue with an ad blocker interceptor. But we are definitely need an `onError` handler.

**Comment by EugeneBalabai** [comment]
@neilkakkar yep, but I also want to show the loader to a customer until I get the values from Posthog to prevent flickering

**Comment by vonBrax** [comment]
I'm facing the same or a similar problem here.

For me the problem is with the [surveys module which uses a `setInterval`](https://github.com/PostHog/posthog-js/blob/0f9a41342da9ac2878ea5572a62aeef3ff2567e9/src/extensions/surveys.tsx#L355) with no break conditions. Every second it tries to load the surveys and check if there is an active matching survey. If an ad blocker is enabled, the call to the surveys endpoint will fail and then retry indefinitely every second. The error logs completely take over the browser console and the network tab, which is really annoying.

![image](https://github.com/user-attachments/assets/0d3ae31c-2962-40d6-a2d1-8a8c0485309d)

I had to add a workaround where I manually check if posthog server is reachable before initializing posthog client.

It would be very nice if posthog could expose an api to do some health checks before the initialisation method is called or perfom the checks on the init method and bail out in case the server can't be reached.
